### PR TITLE
Trail Map: Centralise experiment logic within hook

### PR DESCRIPTION
**Pull request by oswian**
Relates to https://github.com/Automattic/wp-calypso/milestone/476

## Proposed Changes

Cleanup focused PR to simplify/consolidate our experiment-related code.

- Moves experiment logic out of `plans-features-main/index.tsx` and into the hook `use-experiment-for-trail-map.ts`
- Addresses the duplication of business logic between `use-experiment-for-trail-map.ts` and `experiments.ts`
    - Both previously contained the same variant checks:
        - eg. `variant === 'treatment-copy-and-structure' || variant === 'treatment-copy'`
    - Now only `experiments.ts` includes this logic, which is then reused by the hook

## Testing Instructions

This is a refactor with no expected functional changes, so please just smoke test each variant:

- `/start/plans?flags=onboarding/trail-map-feature-grid-copy`
- `/start/plans?flags=onboarding/trail-map-feature-grid-structure`
- `/start/plans?flags=onboarding/trail-map-feature-grid`

## Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-aUh-p2)?

### CODE REVIEW

File: client/my-sites/plans-features-main/hooks/use-experiment-for-trail-map.ts [hunk 1]
```diff
@@ -1,45 +1,47 @@
...
+	isTrailMapStructureVariant,
+	setTrailMapExperiment,
+	TrailMapVariantType as VariantType,
+} from '@automattic/calypso-products';
+import { useEffect } from 'react';
 import { useExperiment } from 'calypso/lib/explat';
-import type { DataResponse } from '@automattic/plans-grid-next';
 
-interface Props {
+interface Params {
```
Comment (oswian) [inline client/my-sites/plans-features-main/hooks/use-experiment-for-trail-map.ts:12]: I tend to prefer the term `Params` for hooks, to differentiate from component `Props`.

File: client/my-sites/plans-features-main/hooks/use-experiment-for-trail-map.ts [hunk 2]
```diff
@@ -1,45 +1,47 @@
...
 	flowName?: string | null;
 }
 
-export const TrailMapVariant = {
-	Control: 'control',
-	TreatmentCopy: 'treatment-copy',
-	TreatmentStructure: 'treatment-structure',
-	TreatmentCopyAndStructure: 'treatment-copy-and-structure',
-} as const;
```
Comment (oswian) [inline client/my-sites/plans-features-main/hooks/use-experiment-for-trail-map.ts:14]: I think we can achieve the same type safety by just using the variant strings and avoiding the need for this object.

**Review by aneeshd16 — CHANGES_REQUESTED** [review]
Love the cleanup! I was actually hoping for this change.

I have some feedback about this:

* The experiment variants (21737-explat-experiment) have underscores instead of dashes. So the actual variants are: `control`, `treatment_copy`, `treatment_copy_and_structure` and `treatment_structure`. This bug was there previously, but this PR is a good place to fix this.
* Since `useExperiment` makes a network call, we do need the `isLoading` value returned by that hook, and that should be patched into `isLoadingGridPlans`: https://github.com/Automattic/wp-calypso/blob/30249a46d040df11e900a50a2d038e46dd5e5a63/client/my-sites/plans-features-main/index.tsx#L701-L703
* We also need this hook to return discrete values for the variants, since that is used here: https://github.com/Automattic/wp-calypso/pull/90195/files#diff-18214fd84316736ff6662f03fa201b6255f4c20f277c0eb683b5e36e9c9dea6d

**Comment by oswian** [comment]
Thanks for the review @aneeshd16. Updated to address your feedback: https://github.com/Automattic/wp-calypso/pull/90256/commits/a09a06a5a6993c7d869c27e7106e7acc5f1f4ce0

**Comment by oswian** [comment]
Hey @aneeshd16, were there any other changes to address here, or ok to ship?

**Review by aneeshd16 — APPROVED** [review]
Thanks for making the changes, LGTM!

**Pull request merged**